### PR TITLE
added CRUD operations for the file system (only in azure table)

### PR DESCRIPTION
**Pull request by IonutPad15**

### CODE REVIEW

**Comment by iiusan** [comment]
feature incomplete

File: IssueTracker/DataAccess/DataAccess.csproj [hunk 1]
```diff
@@ -7,10 +7,13 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Azure.Storage.Blobs" Version="12.13.1" />
```
Comment (iiusan) [inline IssueTracker/DataAccess/DataAccess.csproj:10]: is this needed here?

File: IssueTracker/DataAccess/DataAccess.csproj [hunk 2]
```diff
@@ -7,10 +7,13 @@
...
     <PackageReference Include="Dapper" Version="2.0.123" />
+    <PackageReference Include="Microsoft.Azure.Cosmos.Table" Version="1.0.8" />
```
Comment (iiusan) [inline IssueTracker/DataAccess/DataAccess.csproj:12]: is this needed here?

File: IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿using Models.Request;
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs:1]: dependency on higher level abstractions

File: IssueTracker/IssueTracker.FileSystem/Data/MetaData.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+using Models.Response;
+
+namespace IssueTracker.FileSystem.Data;
+
+public class MetaData : IMetaData
+{
+    private readonly CloudTable _metaDataTable;
+    public MetaData()
+    {
+        var storageAccount = CloudStorageAccount.Parse("AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;DefaultEndpointsProtocol=http;BlobEndpoint=http://127.0.0.1:10000/devstoreaccount1;QueueEndpoint=http://127.0.0.1:10001/devstoreaccount1;TableEndpoint=http://127.0.0.1:10002/devstoreaccount1;");
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Data/MetaData.cs:14]: move into appsettings.json

File: IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+using Models.Response;
+
+namespace IssueTracker.FileSystem.Data.IData;
+
+public interface IMetaData
+{
+    IEnumerable<MetaDataResponse> GetAll();
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs:8]: is GetAll() a valid use case?

File: IssueTracker/IssueTracker.FileSystem/Data/MetaData.cs [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        _metaDataTable = tableClient.GetTableReference("MetaData");
+    }
+    public IEnumerable<MetaDataResponse> GetAll()
+    {
+        var query = new TableQuery<MetaDataEntity>();
+        var entities = _metaDataTable.ExecuteQuery(query);
+        var models = entities.Select(x => new MetaDataResponse(x.RowKey, x.PartitionKey, x.Name, x.Type, x.SizeKb));
+        return models;
+    }
+    public async Task CreateOrUpdateAsync(MetaDataRequest entity)
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Data/MetaData.cs:25]: no validation! what happens if a field is empty? merge or replace?

File: IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+    Task CreateOrUpdateAsync(MetaDataRequest entity);
+    Task<bool> DeleteAsync(string id, string group);
+    Task<MetaDataResponse?> GetAsync(string rowKey, string partitionKey);
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs:11]: RowKey and PartitionKey are implementation details and should not be exposed here

File: IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs [hunk 8]
```diff
@@ -0,0 +1,12 @@
+﻿using Models.Request;
+using Models.Response;
+
+namespace IssueTracker.FileSystem.Data.IData;
+
+public interface IMetaData
+{
+    IEnumerable<MetaDataResponse> GetAll();
+    Task CreateOrUpdateAsync(MetaDataRequest entity);
+    Task<bool> DeleteAsync(string id, string group);
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs:10]: what do 'id' and 'group' mean?

File: IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs [hunk 9]
```diff
@@ -0,0 +1,12 @@
+﻿using Models.Request;
+using Models.Response;
+
+namespace IssueTracker.FileSystem.Data.IData;
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs:4]: simplify the namespace: IssueTracker.FileSystem

File: IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs (hunk 5 above)
Comment (IonutPad15) [reply IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs:8]: not really

File: IssueTracker/DataAccess/DataAccess.csproj (hunk 1 above)
Comment (IonutPad15) [reply IssueTracker/DataAccess/DataAccess.csproj:10]: no, I forgot to delete

File: IssueTracker/DataAccess/DataAccess.csproj (hunk 2 above)
Comment (IonutPad15) [reply IssueTracker/DataAccess/DataAccess.csproj:12]: no, I forgot to delete

File: IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs (hunk 3 above)
Comment (IonutPad15) [reply IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs:1]: I changed

File: IssueTracker/IssueTracker.FileSystem/Data/MetaData.cs (hunk 4 above)
Comment (IonutPad15) [reply IssueTracker/IssueTracker.FileSystem/Data/MetaData.cs:14]: I moved

File: IssueTracker/IssueTracker.FileSystem/Data/MetaData.cs (hunk 6 above)
Comment (IonutPad15) [reply IssueTracker/IssueTracker.FileSystem/Data/MetaData.cs:25]: I validated

File: IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs (hunk 7 above)
Comment (IonutPad15) [reply IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs:11]: I changed

File: IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs (hunk 8 above)
Comment (IonutPad15) [reply IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs:10]: the rowkey and partition key, but the API doesn't have to know that

File: IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs (hunk 9 above)
Comment (IonutPad15) [reply IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs:4]: I simplified

File: IssueTracker/IssueTracker.FileSystem/Data/MetaData.cs (hunk 4 above)
Comment (iiusan) [reply IssueTracker/IssueTracker.FileSystem/Data/MetaData.cs:14]: remove the appsettings.json from the FileSystem project

File: IssueTracker/IssueTracker.FileSystem/Data/MetaData.cs (hunk 6 above)
Comment (iiusan) [reply IssueTracker/IssueTracker.FileSystem/Data/MetaData.cs:25]: where is the validation? line #

File: IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs (hunk 9 above)
Comment (iiusan) [reply IssueTracker/IssueTracker.FileSystem/Data/IData/IMetaData.cs:4]: not everywhere

**Pull request closed without merging**